File: backend/models/resume.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Dict, Any
from datetime import datetime

class ResumeBase(BaseModel):
    # Core Identity (same as profile)
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    
    # Career Profile (same as profile)
    professional_summary: Optional[str] = None
    career_level: Optional[str] = None
    years_experience: Optional[int] = None
    primary_domain: Optional[str] = None
    
    # JSON fields for flexible data storage (same as profile)
    seniority_keywords: Optional[Dict[str, Any]] = None
    experience: Optional[Dict[str, Any]] = None
    education: Optional[Dict[str, Any]] = None
    skills: Optional[Dict[str, Any]] = None
    languages: Optional[Dict[str, Any]] = None
    
    # Enhanced Profile Data (same as profile)
    career_trajectory: Optional[Dict[str, Any]] = None
    domain_expertise: Optional[Dict[str, Any]] = None
    leadership_experience: Optional[Dict[str, Any]] = None
    achievement_highlights: Optional[Dict[str, Any]] = None
    
    # Resume specific fields
    source_documents: Optional[Dict[str, Any]] = None  # references to profile version used
    misc_data: Optional[Dict[str, Any]] = None
    
    # File storage (local for now, S3 later)
    file_path: Optional[str] = None  # local path to generated/uploaded PDF (will become S3 URL)
    file_type: str = "generated"  # generated, uploaded
    
    # Resume metadata
    version: int = 1  # for tracking resume versions
    is_active: bool = True
    customization_notes: Optional[str] = None  # notes about customizations for specific job

class Resume(ResumeBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    profile_id: int
    job_posting_id: Optional[int] = None  # optional link to specific job posting
    created_at: datetime
    updated_at: datetime

class ResumeCreate(BaseModel):
    # Required fields for creation
    name: str
    profile_id: int
    
    # Optional fields for creation
    job_posting_id: Optional[int] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    professional_summary: Optional[str] = None
    career_level: Optional[str] = None
    years_experience: Optional[int] = None
    primary_domain: Optional[str] = None
    seniority_keywords: Optional[Dict[str, Any]] = None
    experience: Optional[Dict[str, Any]] = None
    education: Optional[Dict[str, Any]] = None
    skills: Optional[Dict[str, Any]] = None
    languages: Optional[Dict[str, Any]] = None
    career_trajectory: Optional[Dict[str, Any]] = None
    domain_expertise: Optional[Dict[str, Any]] = None
    leadership_experience: Optional[Dict[str, Any]] = None
    achievement_highlights: Optional[Dict[str, Any]] = None
    source_documents: Optional[Dict[str, Any]] = None
    misc_data: Optional[Dict[str, Any]] = None
    file_path: Optional[str] = None
    file_type: str = "generated"
    version: int = 1
    is_active: bool = True
    customization_notes: Optional[str] = None

class ResumeUpdate(BaseModel):
    # All fields optional for updates
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    professional_summary: Optional[str] = None
    career_level: Optional[str] = None
    years_experience: Optional[int] = None
    primary_domain: Optional[str] = None
    seniority_keywords: Optional[Dict[str, Any]] = None
    experience: Optional[Dict[str, Any]] = None
    education: Optional[Dict[str, Any]] = None
    skills: Optional[Dict[str, Any]] = None
    languages: Optional[Dict[str, Any]] = None
    career_trajectory: Optional[Dict[str, Any]] = None
    domain_expertise: Optional[Dict[str, Any]] = None
    leadership_experience: Optional[Dict[str, Any]] = None
    achievement_highlights: Optional[Dict[str, Any]] = None
    source_documents: Optional[Dict[str, Any]] = None
    misc_data: Optional[Dict[str, Any]] = None
    file_path: Optional[str] = None
    file_type: Optional[str] = None
    version: Optional[int] = None
    is_active: Optional[bool] = None
    customization_notes: Optional[str] = None

class ResumeRead(Resume):
    pass

File: frontend/src/types/resume.ts
export interface Job {
  title: string
  years: string
  company: string
  location: string
  description: string
}

export interface Degree {
  gpa?: string
  years: string
  degree: string
  courses?: string
  location: string
  university: string
}

// Profile数据结构（完整的用户档案）
export interface ProfileData {
  name: string
  email: string
  phone: string
  location: string
  open_to_relocate?: boolean
  professional_summary?: string
  career_level?: string | null
  years_experience?: number
  primary_domain?: string | null
  seniority_keywords?: string | null
  experience?: {
    jobs: Job[]
  }
  education?: {
    degrees: Degree[]
  }
  skills?: {
    raw_skills: string[]
  }
  languages?: {
    spoken: string[]
  }
  career_trajectory?: any
  domain_expertise?: any
  leadership_experience?: any
  achievement_highlights?: any
  source_documents?: {
    original_resume: string
  }
  processing_quality?: number
  last_resume_update?: any
  processing_history?: any
  enhancement_status?: string
  confidence_score?: any
  data_sources?: {
    sources: string[]
  }
  keywords?: any
  completeness_metadata?: any
  misc_data?: any
  notes?: any
  id?: number
  user_id?: number
  created_at?: string
  updated_at?: string
}

// Resume数据结构（简历特定数据）
export interface ResumeData {
  name: string
  email: string
  phone: string
  location?: string | null
  professional_summary?: string
  career_level?: string | null
  years_experience?: number
  primary_domain?: string | null
  seniority_keywords?: string | null
  experience?: any
  education?: {
    degrees: Degree[]
  }
  skills?: {
    raw_skills: string[]
  }
  languages?: any
  career_trajectory?: any
  domain_expertise?: any
  leadership_experience?: any
  achievement_highlights?: any
  source_documents?: any
  misc_data?: any
  file_path?: string | null
  file_type?: string
  version?: number
  is_active?: boolean
  customization_notes?: any
  id?: number
  profile_id?: number
  job_posting_id?: number | null
  created_at?: string
  updated_at?: string
}

// API响应结构
export interface ParseResumeResponse {
  success: boolean
  profile?: ProfileData
  resume?: ResumeData
  error?: string
  message?: string
}

// 向后兼容的ExtractedData类型（基于ProfileData）
export type ExtractedData = ProfileData & {
  // 兼容旧版本字段
  title?: string
  linkedin?: string
  sponsorship?: string
}

export interface PersonalExtra {
  label: string
  value: string
}

export interface EducationExtra {
  university: string
  degreeType: string
  major: string
}
